libraries/chain/include/eosio/chain/types.hpp

```diff
@@ -410,20 +409,78 @@ namespace eosio::chain {
 
 }  // eosio::chain
 
+
+namespace fc::raw {
+   // pack/unpack chainbase::shared_cow_string
+   // ----------------------------------------
+   template<typename Stream>
+   inline void pack(Stream& s, const chainbase::shared_string& v) {		
+      FC_ASSERT(v.size() <= MAX_SIZE_OF_BYTE_ARRAYS);		
+      fc::raw::pack(s, fc::unsigned_int((uint32_t)v.size()));		
+      if (v.size())
+         s.write(v.data(), v.size());		
+   }		
+
+   template<typename Stream>
+   inline void unpack(Stream& s, chainbase::shared_string& v) {
```

Comment:
I realize that unpack of `std::string` does not check `MAX_SIZE_OF_BYTE_ARRAYS` but I wonder if that is a mistake. Also why are we enforcing these limits at all?